docs/en/reference/query-builder.rst

```diff
@@ -9,6 +9,11 @@ programmatically build queries, and also provides a fluent API.
 This means that you can change between one methodology to the other
 as you want, or just pick a preferred one.
 
+.. note::
+
+    The ``QueryBuilder`` is not an abstraction of DQL, but merely a tool to dynamically build it.
+    You should still use plain DQL when you can, as it is simpler and more readable.
```

Comment:
How about this?

```suggestion
    Dynamically built DQL queries usually are the exception though, not
    the rule. For the majority of queries that do not need to change
    based on some condition, you should default to using DQL strings,
    because Doctrine would always build a DQL string behind the scenes
    anyway, and because it is simpler and more readable.
```

Reply:
While your text is fine, I'm hesitant because I don't want to over-explain it, because then people might not read it or read it and fail to get the point. Maybe I can add this to the FAQ part instead, because I don't think people will have the required attention span here. IMHO it's better to write "just STFU and do this because it's better" here, which is what most people want to read in the documentation. And for people who really want to know why or settle a debate between developer gurus, there is the FAQ.

Reply:
Agree, the explanation is fine, but going straightforward and appending `For more details, go there(FAQ link)`

Reply:
You might also put a link to the section of the faq into this note.

Reply:
Actually your text didn't fit well with my paragraph in the FAQ so I wrote it differently.
@greg0ire please tell me if the new paragraph in the FAQ covers your point.